presto-docs/src/main/sphinx/optimizer/push-down.rst

```diff
@@ -0,0 +1,140 @@
+
+The results of this push down can include the following benefits:
+
+* improved overall query performance
+* reduced network traffic between Presto and the database
+* reduced load on the database
+
+Support for push down is specific to each connector and the relevant underlying
+database or storage system.
+
+Every Presto aggregate function is eligible for push down, as long as it
+operates on types supported by the given connector.
+
+Analysis and Confirmation
+-------------------------
+
+Push down depends on a number of factors:
+
+* generic support for push down in the connector
+* function or operation specific support for push down in the connector
+* query that allows the detection of the function to push down
+
+The best way to analyze if push down for a specific query is performed, is to
+take a closer look at the :doc:`EXPLAIN plan </sql/explain>` of the query. If an
+operations such as an aggregate function successfully pushed down to the
+connector, the explain plan does **not** show that operator. The explain plan
+only shows the operations that are performed by Presto.
+
+Here is an example query using the TPCH data, which was pushed into a PostgreSQL
+database and it now queried with the PostgreSQL connector::
+
+    USE postgresql.tpch.tiny;
+    SELECT regionkey, count(*)
+    FROM nation
+    GROUP BY 1;
+
+You can get the explain plan with the following statement::
+
+    EXPLAIN
+    SELECT regionkey, count(*)
+    FROM nation
+    GROUP BY 1;
+
+The explain plan for this query does not show any ``count`` operation as
+``Aggregate`` in Presto. This is the result of a successful push down.
+
+.. code-block:: none
+
+    Fragment 0 [SINGLE]
+        Output layout: [regionkey_0, _presto_generated_1]
+        Output partitioning: SINGLE []
+        Stage Execution Strategy: UNGROUPED_EXECUTION
+        Output[regionkey, _col1]
+        │   Layout: [regionkey_0:bigint, _presto_generated_1:bigint]
+        │   Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: ?}
+        │   regionkey := regionkey_0
+        │   _col1 := _presto_generated_1
+        └─ RemoteSource[1]
+                Layout: [regionkey_0:bigint, _presto_generated_1:bigint]
+
+    Fragment 1 [SOURCE]
+        Output layout: [regionkey_0, _presto_generated_1]
+        Output partitioning: SINGLE []
+        Stage Execution Strategy: UNGROUPED_EXECUTION
+        TableScan[postgresql:tpch.nation tpch.nation columns=[regionkey:bigint:int8, count(*):_presto_generated_1:bigint:bigint] groupingSets=[[regionkey:bigint:int8]], gro
+            Layout: [regionkey_0:bigint, _presto_generated_1:bigint]
+            Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: 0B}
+            _presto_generated_1 := count(*):_presto_generated_1:bigint:bigint
+            regionkey_0 := regionkey:bigint:int8
+
+A number of factors can prevent a push down:
+
+* adding a condition to the query
+* using a different aggregate function without push down support
```

Comment:
As above, the last two items seem similar. How about rephrasing as
```
A number of factors can prevent pushdown, including adding a ``WHERE`` clause
condition or using an aggregation function not supported by the connector.

Reply:
Reworded but left as list since this is clearer.